### PR TITLE
Add connected charts

### PR DESCRIPTION
**Pull request by hughess**
### Description
Adds `connectGroup` prop (string) which connects any charts sharing the same `connectGroup` and provides synchronized tooltip hovering.

https://github.com/evidence-dev/evidence/assets/12602440/3f875873-9c61-49f0-854b-35a1d9ea5926

CleanShot 2024-03-24 at 19 55 59

### To do before merging
- Test connection on Heatmap, CalendarHeatmap
- Look into interaction options on map - highlighting hovered state and showing value in legend would be nice

### Checklist
- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable

### CODE REVIEW

File: packages/ui/core-components/src/lib/unsorted/viz/core/ECharts.svelte [hunk 1]
```diff
@@ -183,7 +187,6 @@
 	.chart {
 		-moz-user-select: none;
 		-webkit-user-select: none;
-		-ms-user-select: none;
```
Comment (csjh) [inline packages/ui/core-components/src/lib/unsorted/viz/core/ECharts.svelte:186]: Why was this removed?

File: packages/lib/component-utilities/src/echarts.js
Comment (csjh) [inline packages/lib/component-utilities/src/echarts.js:1]: `echarts.js` and `echartsMap.js` should probably be merged (out of scope of this PR)

**Review by csjh — APPROVED** [review]

File: packages/ui/core-components/src/lib/unsorted/viz/core/ECharts.svelte (hunk 1 above)
Comment (hughess) [reply packages/ui/core-components/src/lib/unsorted/viz/core/ECharts.svelte]: that i'm not sure - wasn't intentional. just added it back
